### DISCUSSION TITLE
Apollo server (stop + start)

### DISCUSSION THREAD

**Discussion by hugoribeiromsglife** [apollographql/apollo-server#7913] [Q&A]
Hey, hope you are doing great 👍 

I have been using Apollo Server with an http server as documented here https://www.apollographql.com/docs/apollo-server/api/plugin/drain-http-server/. I am also using the drain plugin mentioned there.
I have a feature that dynamically adds/deletes servers at runtime, meaning: I can have a server running at /xyz, delete it (closing http server and stopping apollo server: 'httpServer.closeAllConnections(); httpServer.close(); await apolloServer.stop();') and then try to create it again at runtime, but.. when I do that I always face the following exception:

"A GraphQL operation was received during server shutdown. The operation will fail. Consider draining the HTTP server on shutdown; see https://go.apollo.dev/s/drain for details."

any hints here?

Thanks and have a nice day!

**Comment by pepew-le-boss** [comment]
same here, have you found a solution ?

**Reply by hugoribeiromsglife** [reply]
Not yet.. we focused on other things we have to do and decreased the priority on this one.

**Comment by MukundaKatta** [comment]
The race is between `apolloServer.stop()` returning and the server being ready to accept a new start. Apollo Server v4 doesn't support stop→start on the same instance — you need a fresh `new ApolloServer(...)` after teardown. Track your servers in a `Map<path, ApolloServer>`, call `stop()`, `delete` the entry, then `new ApolloServer()` + `start()` for the new one. Reusing the old instance triggers exactly that shutdown error.
